### PR TITLE
Remove `test_private_bucket` object_store test

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-rs/issues/6600

# Rationale for this change
The test is failing on main

I think it is due to whatever the `bloxblender` s3 bucket being moved 

# What changes are included in this PR?

Remove the test as it is testing some behavior of S3 (see @tustvold 's comment below)

# Are there any user-facing changes?

No this is a test only

### CODE REVIEW

File: object_store/src/aws/resolve.rs [hunk 1]
```diff
@@ -80,13 +80,13 @@ mod tests {
 
     #[tokio::test]
     async fn test_private_bucket() {
-        let bucket = "bloxbender";
+        let bucket = "public-datasets";
```
Comment (alamb) [inline object_store/src/aws/resolve.rs:83]: the test is called `private-bucket` but then uses no authentication. I don't know what to make of this.

Maybe @tustvold  remembers

Comment (tustvold) [reply object_store/src/aws/resolve.rs:83]: IIRC it was a test to make sure resolve_region worked for private buckets without needing credentials.

**Comment by tustvold** [comment]
I think we can probably just remove this test, its effectively just testing a behaviour of S3 which isn't going to change

**Comment by alamb** [comment]
> I think we can probably just remove this test, its effectively just testing a behaviour of S3 which isn't going to change

removed in 46a8e0a

**Review by tustvold — APPROVED** [review]

**Review by andygrove — APPROVED** [review]

File: object_store/src/aws/resolve.rs [hunk 2]
```diff
@@ -78,19 +78,6 @@ pub async fn resolve_bucket_region(bucket: &str, client_options: &ClientOptions)
 mod tests {
     use super::*;
 
-    #[tokio::test]
-    async fn test_private_bucket() {
```
Comment (findepi) [inline object_store/src/aws/resolve.rs:82]: the PR description says "Fix test_private_bucket object_store test"
and "Change to use another s3 bucket"

let's maybe update both before merge

Comment (alamb) [reply object_store/src/aws/resolve.rs:82]: done. Thank you

**Pull request merged**